spring-cloud-kubernetes-ribbon/src/main/java/org/springframework/cloud/kubernetes/ribbon/KubernetesServicesServerList.java

```diff
@@ -76,7 +76,7 @@ private String concatServiceFQDN(Service service) {
 			}
 			else {
 				for (ServicePort servicePort : service.getSpec().getPorts()) {
-					if (Utils.isNotNullOrEmpty(this.getPortName())
```

Comment:
It seems like this should be `if (Utils.isNotNullOrEmpty(this.getPortName()) && this.getPortName().endsWith(servicePort.getName()))`

Maybe a unit test would help clarify things?

Reply:
Hi, I have add a unit test to explain this.